### PR TITLE
Bi 10152 extract corp dir details

### PR DESCRIPTION
**Pull request by mattch1**
Extract and display the corporate director fields.

### CODE REVIEW

File: views/tasks/active-officers-details.html [hunk 1]
```diff
@@ -41,62 +41,7 @@ <h1 class="govuk-heading-l">Check the officers' details</h1>
...
-              <div class="govuk-summary-list__row">
-                <dt class="govuk-summary-list__key">Registration number</dt>
-                <dd class="govuk-summary-list__value govuk-!-padding-top-0 govuk-!-padding-bottom-0">
-                  <p>0111222</p>
-                </dd>
-              </div>
-            </dl>
-          </div>
-        </div>
+           {% include "includes/corporate-directors.html" %}
```
Comment (markpit) [inline views/tasks/active-officers-details.html:44]: need to add the empty list check before inserting the corpDirectorsHTML into the rows array

File: views/includes/corporate-directors.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% if corporateDirectorList.length > 1 %}
+<h2 class="govuk-heading-m">{{ corporateDirectorList.length }} corporate directors</h2>
```
Comment (markpit) [inline views/includes/corporate-directors.html:2]: the if else could do with indenting

File: views/tasks/active-officers-details.html [hunk 3]
```diff
@@ -41,62 +41,7 @@ <h1 class="govuk-heading-l">Check the officers' details</h1>
...
         {% endset %}
 
       {% set officerRows = [] %}
```
Comment (mwestacott) [inline views/tasks/active-officers-details.html:47]: I think you need that 'if' block now around the 'set officerRows' statement for corpDirectors below. Missing for directorHTML too it seems.

File: views/includes/corporate-secretaries.html [hunk 4]
```diff
@@ -1,71 +1,9 @@
 {% if corporateSecretaryList.length > 1 %}
-  <h2 class="govuk-heading-m">{{ corporateSecretaryList.length }} secretaries</h2>
+  <h2 class="govuk-heading-m">{{ corporateSecretaryList.length }} corporate secretaries</h2>
 {% elseif corporateSecretaryList.length === 1 %}
   <h2 class="govuk-heading-m">1 corporate secretary</h2>
 {% endif %}
 
-{% for secretary in corporateSecretaryList %}
-  <div class="govuk-summary-card govuk-!-margin-bottom-4">
```
Comment (mwestacott) [inline views/includes/corporate-secretaries.html:8]: Nice that we've got rid of all this duplicated code.

File: test/controllers/tasks/active.officers.details.controller.unit.ts [hunk 5]
```diff
@@ -68,12 +68,23 @@ describe("Active officers details controller tests", () => {
 
       expect(mockGetActiveOfficerDetails).toHaveBeenCalled();
       expect(response.text).toContain("MYERS, Micheal");
-      expect(response.text).toContain("Diddly Squat Farm Shop, Chadlington, Thisshire, England, OX7 3PE");
-      expect(response.text).toContain("1 January 2009");
+      expect(response.text).toContain("10 Corpsecret Road, Corpsecrettown, Corpsecretshire, Corpsecretland, CS1 7SC");
```
Comment (mwestacott) [inline test/controllers/tasks/active.officers.details.controller.unit.ts:71]: Wondering why the corp secretary test changed with this commit? Corp secretaries in theory aren't even secure, so the test data is confusing me a bit too.

File: views/includes/corporate-officer.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            {% if officer.placeRegistered %}
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">Place registered</dt>
+                <dd class="govuk-summary-list__value govuk-!-padding-top-0 govuk-!-padding-bottom-0">
+                    <p>{{officer.placeRegistered}}</p>
+                </dd>
+            </div>
+            {% endif %}
+            {% if officer.registrationNumber %}
+            <div class="govuk-summary-list__row">
```
Comment (SVarallo) [inline views/includes/corporate-officer.html:54]: Small thing but the indentation in these if statements

File: test/controllers/tasks/active.officers.details.controller.unit.ts (hunk 5 above)
Comment (mattch1) [reply test/controllers/tasks/active.officers.details.controller.unit.ts:71]: We want to make sure that the test data is unique for each offficer beacuse of the way these are being tested so needed to be changed.

Comment (mwestacott) [reply test/controllers/tasks/active.officers.details.controller.unit.ts:71]: Aha - I get it now! Was thinking 'Corpsecret' was representing something 'secret' (for a secure secretary). If it confused me, could confuse others so would suggest changing to not use a shortened form of 'secretary' but up to you.

**Review by mwestacott — APPROVED** [review]
Approving but please see response to your comment @mattch1 .

**Pull request merged**